File: internal/commands/help.go
package commands

import (
	"errors"
	"fmt"
	"io"
	"strings"
	"text/tabwriter"

	"github.com/spf13/cobra"
)

type configKeyDoc struct {
	Key         string
	Type        string
	Default     string
	Description string
	Example     string
	Impact      string
}

var configDocs = []configKeyDoc{
	{
		Key:         "adr_dirs",
		Type:        "list of strings",
		Default:     `["docs/adr"]`,
		Description: "Diretórios onde os ADRs são armazenados.",
		Example:     `adr_dirs:
  - docs/adr
  - docs/adr/zeus`,
		Impact: "Alterar adiciona ou remove diretórios rastreados nas validações de ADR.",
	},
	{
		Key:         "req_dir",
		Type:        "string",
		Default:     `"docs/req"`,
		Description: "Diretório onde as REQs são armazenadas.",
		Example:     `req_dir: docs/requirements`,
		Impact:      "Alterar muda onde o gate procura REQs para validação.",
	},
	{
		Key:         "roadmap_dir",
		Type:        "string",
		Default:     `"docs/roadmaps"`,
		Description: "Diretório raiz dos roadmaps.",
		Example:     `roadmap_dir: docs/roadmaps`,
		Impact:      "Alterar muda onde o gate procura roadmaps em backlog/wip/blocked/done.",
	},
	{
		Key:         "roadmap_namespacing",
		Type:        "flat|by_agent",
		Default:     `"flat"`,
		Description: "Estratégia de namespacing dos roadmaps.",
		Example:     `roadmap_namespacing: by_agent`,
		Impact:      "by_agent cria subdiretórios por agente; flat usa diretório único por estado.",
	},
	{
		Key:         "agents",
		Type:        "list of strings",
		Default:     `[]`,
		Description: "Lista de agentes ativos no projeto.",
		Example: `agents:
  - apolo
  - afrodite`,
		Impact: "Agentes registrados recebem subdiretórios próprios no modo by_agent.",
	},
	{
		Key:         "governance_mode",
		Type:        "string",
		Default:     `""`,
		Description: "Modo de governança (strict, lenient).",
		Example:     `governance_mode: lenient`,
		Impact:      "lenient converte violations em warnings; strict (padrão) bloqueia com exit 1.",
	},
	{
		Key:         "lenient_until",
		Type:        "date (YYYY-MM-DD)",
		Default:     `""`,
		Description: "Data até quando o modo lenient está ativo.",
		Example:     `lenient_until: 2026-12-31`,
		Impact:      "Após a data, o modo volta a strict automaticamente.",
	},
	{
		Key:         "wip_limit",
		Type:        "integer",
		Default:     `1`,
		Description: "Limite de itens WIP simultâneos.",
		Example:     `wip_limit: 3`,
		Impact:      "Aumentar reduz a frequência de bloqueio; diminuir exige mais disciplina.",
	},
	{
		Key:         "wip_by_squad",
		Type:        "boolean",
		Default:     `false`,
		Description: "Aplicar limite WIP por squad individualmente.",
		Example:     `wip_by_squad: true`,
		Impact:      "true aplica o limite por squad; false aplica ao total do projeto.",
	},
	{
		Key:         "require_req_in_commit",
		Type:        "boolean",
		Default:     `false`,
		Description: "Exigir referência de REQ em mensagens de commit.",
		Example:     `require_req_in_commit: true`,
		Impact:      "true instala hook commit-msg que bloqueia commits sem referência REQ.",
	},
	{
		Key:         "link_fields.req",
		Type:        "list of strings",
		Default:     `["REQ:"]`,
		Description: "Marcadores que identificam link a REQ.",
		Example: `link_fields:
  req:
    - "REQ:"
    - "req_id:"`,
		Impact: "Alterar muda quais tokens o gate reconhece como link de REQ.",
	},
	{
		Key:         "link_fields.adr",
		Type:        "list of strings",
		Default:     `["ADR:"]`,
		Description: "Marcadores que identificam link a ADR.",
		Example: `link_fields:
  adr:
    - "ADR:"
    - "adr_id:"`,
		Impact: "Alterar muda quais tokens o gate reconhece como link de ADR.",
	},
	{
		Key:         "link_fields.roadmap",
		Type:        "list of strings",
		Default:     `["Roadmap:"]`,
		Description: "Marcadores que identificam link a Roadmap.",
		Example: `link_fields:
  roadmap:
    - "Roadmap:"
    - "roadmap_id:"`,
		Impact: "Alterar muda quais tokens o gate reconhece como link de Roadmap.",
	},
	{
		Key:         "acceptance_markers",
		Type:        "list of strings",
		Default:     `["## Acceptance Criteria", "## Critérios de Aceite"]`,
		Description: "Marcadores de critério de aceite em documentos WIP.",
		Example: `acceptance_markers:
  - "## Acceptance Criteria"
  - "## Critérios de Aceite"
  - "## AC"`,
		Impact: "Alterar permite personalizar o cabeçalho de seção que o gate valida.",
	},
	{
		Key:         "rules.wip_has_req",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Severidade: WIP sem REQ linkada.",
		Example:     `rules:\n  wip_has_req: warning`,
		Impact:      "error bloqueia o gate; warning exibe mas não bloqueia; off ignora.",
	},
	{
		Key:         "rules.wip_acceptance",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Severidade: WIP sem critérios de aceite.",
		Example:     `rules:\n  wip_acceptance: warning`,
		Impact:      "error bloqueia o gate; warning exibe mas não bloqueia; off ignora.",
	},
	{
		Key:         "rules.wip_limit",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Severidade: excesso de itens WIP.",
		Example:     `rules:\n  wip_limit: warning`,
		Impact:      "error bloqueia o gate; warning exibe mas não bloqueia; off ignora.",
	},
	{
		Key:         "rules.stale_wip",
		Type:        "off|warning|error",
		Default:     `"warning"`,
		Description: "Severidade: WIP sem atualização recente.",
		Example:     `rules:\n  stale_wip: error`,
		Impact:      "Aumentar severidade força revisão de roadmaps parados.",
	},
	{
		Key:         "rules.adr_orphan",
		Type:        "off|warning|error",
		Default:     `"warning"`,
		Description: "Severidade: ADR sem REQ vinculada.",
		Example:     `rules:\n  adr_orphan: error`,
		Impact:      "error força que todo ADR seja referenciado por ao menos uma REQ.",
	},
	{
		Key:         "rules.ref_targets_exist",
		Type:        "off|warning|error",
		Default:     `"warning"`,
		Description: "Severidade: referências com destino inexistente.",
		Example:     `rules:\n  ref_targets_exist: error`,
		Impact:      "error bloqueia quando REQ ou ADR referenciados não existem no repositório.",
	},
	{
		Key:         "rules.folder_status",
		Type:        "off|warning|error",
		Default:     `"warning"`,
		Description: "Severidade: coerência entre pasta e status do arquivo.",
		Example:     `rules:\n  folder_status: error`,
		Impact:      "error bloqueia quando o status no frontmatter não bate com a pasta (ex: done/ com status Open).",
	},
	{
		Key:         "rules.filename_uniqueness",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Severidade: nomes de arquivo duplicados.",
		Example:     `rules:\n  filename_uniqueness: warning`,
		Impact:      "error bloqueia quando dois artefatos têm o mesmo basename independentemente de pasta.",
	},
	{
		Key:         "rules.blocked_by_draft_adr",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Severidade: REQ bloqueada por ADR em rascunho.",
		Example:     `rules:\n  blocked_by_draft_adr: warning`,
		Impact:      "error bloqueia quando REQ Open referencia ADR com Status: Draft.",
	},
	{
		Key:         "trace_id_field",
		Type:        "string",
		Default:     `""`,
		Description: "Campo de frontmatter usado como ID de rastreabilidade estável entre REQ e Roadmap. Vazio = desativado.",
		Example:     `trace_id_field: req_id`,
		Impact:      "Ativa verificação bidirecional REQ↔Roadmap (traceid_orphan_*, traceid_state_mismatch, traceid_duplicate_*).",
	},
	{
		Key:         "rules.traceid_orphan_roadmap",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Roadmap com req_id sem REQ correspondente.",
		Example:     "rules:\n  traceid_orphan_roadmap: warning",
		Impact:      "Detecta Roadmaps sem REQ pareada.",
	},
	{
		Key:         "rules.traceid_orphan_req",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "REQ com req_id sem Roadmap correspondente.",
		Example:     "rules:\n  traceid_orphan_req: warning",
		Impact:      "Detecta REQs sem Roadmap pareado.",
	},
	{
		Key:         "rules.traceid_state_mismatch",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "REQ e Roadmap com mesmo req_id em estados divergentes (ex: done/wip).",
		Example:     "rules:\n  traceid_state_mismatch: warning",
		Impact:      "Garante consistência de estado entre REQ e Roadmap.",
	},
	{
		Key:         "rules.traceid_duplicate_req",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Mesmo req_id em mais de uma REQ.",
		Example:     "rules:\n  traceid_duplicate_req: warning",
		Impact:      "Garante unicidade lógica de REQs.",
	},
	{
		Key:         "rules.traceid_duplicate_roadmap",
		Type:        "off|warning|error",
		Default:     `"error"`,
		Description: "Mesmo req_id em mais de um Roadmap.",
		Example:     "rules:\n  traceid_duplicate_roadmap: warning",
		Impact:      "Garante unicidade lógica de Roadmaps.",
	},
}

// printCommandList escreve a lista de comandos disponíveis de root em out.
func printCommandList(out io.Writer, root *cobra.Command) {
	w := tabwriter.NewWriter(out, 0, 0, 2, ' ', 0)
	fmt.Fprintln(w, "Comandos disponíveis:")
	for _, sub := range root.Commands() {
		if !sub.IsAvailableCommand() {
			continue
		}
		fmt.Fprintf(w, "  %s\t%s\n", sub.Name(), sub.Short)
	}
	w.Flush()
}

// printConfigKeyList escreve a tabela KEY/DEFAULT/DESCRIÇÃO em out.
func printConfigKeyList(out io.Writer) {
	w := tabwriter.NewWriter(out, 0, 0, 2, ' ', 0)
	fmt.Fprintln(w, "Chaves de configuração (trackfw.yaml):")
	fmt.Fprintln(w, "KEY\tDEFAULT\tDESCRIÇÃO")
	fmt.Fprintln(w, strings.Repeat("─", 80))
	for _, d := range configDocs {
		fmt.Fprintf(w, "%s\t%s\t%s\n", d.Key, d.Default, d.Description)
	}
	w.Flush()
}

// printConfigKeyDoc escreve a documentação completa de uma chave em out.
func printConfigKeyDoc(out io.Writer, d configKeyDoc) {
	fmt.Fprintf(out, "%s\n", d.Key)
	fmt.Fprintf(out, "  Type:    %s\n", d.Type)
	fmt.Fprintf(out, "  Default: %s\n", d.Default)
	fmt.Fprintf(out, "  Desc:    %s\n", d.Description)
	fmt.Fprintf(out, "  Example:\n")
	for _, line := range strings.Split(d.Example, "\n") {
		fmt.Fprintf(out, "    %s\n", line)
	}
	fmt.Fprintf(out, "  Impact:  %s\n", d.Impact)
}

// levenshtein calcula a distância de edição entre a e b.
func levenshtein(a, b string) int {
	ra, rb := []rune(a), []rune(b)
	la, lb := len(ra), len(rb)
	prev := make([]int, lb+1)
	curr := make([]int, lb+1)
	for j := 0; j <= lb; j++ {
		prev[j] = j
	}
	for i := 1; i <= la; i++ {
		curr[0] = i
		for j := 1; j <= lb; j++ {
			cost := 1
			if ra[i-1] == rb[j-1] {
				cost = 0
			}
			min := curr[j-1] + 1
			if prev[j]+1 < min {
				min = prev[j] + 1
			}
			if prev[j-1]+cost < min {
				min = prev[j-1] + cost
			}
			curr[j] = min
		}
		prev, curr = curr, prev
	}
	return prev[lb]
}

// suggestTopic retorna o comando ou chave de configuração conhecida mais
// próxima de topic (distância de edição <= 3), ou "" se nenhuma for próxima
// o suficiente. Usado para compor a mensagem de erro de assunto desconhecido.
func suggestTopic(topic string, root *cobra.Command) string {
	const threshold = 3
	best := ""
	bestDist := threshold + 1

	candidates := make([]string, 0)
	for _, sub := range root.Commands() {
		if sub.IsAvailableCommand() {
			candidates = append(candidates, sub.Name())
		}
	}
	for _, d := range configDocs {
		candidates = append(candidates, d.Key)
	}

	for _, c := range candidates {
		dist := levenshtein(topic, c)
		if dist < bestDist {
			bestDist = dist
			best = c
		}
	}
	if bestDist > threshold {
		return ""
	}
	return best
}

func newHelpCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "help [comando|chave]",
		Short: "Exibe ajuda de comandos e documentação das chaves de configuração do trackfw.yaml",
		Long: `Sem argumento: lista os comandos disponíveis e as chaves configuráveis do trackfw.yaml.
Com argumento: mostra a ajuda do comando informado, ou a documentação completa da
chave de configuração especificada.`,
		Args:               cobra.MaximumNArgs(1),
		DisableFlagParsing: false,
		// Silenciados: a mensagem de erro é composta e impressa por nós mesmos
		// (ver caso 3 abaixo), para que a saída seja idêntica à de Node/Python
		// — sem o bloco "Error: ...\nUsage: ..." que o cobra imprimiria por
		// padrão, o que duplicaria a mensagem quando o processo top-level
		// (Execute em root.go) também reimprime o erro retornado.
		SilenceUsage:  true,
		SilenceErrors: true,
		RunE: func(cmd *cobra.Command, args []string) error {
			out := cmd.OutOrStdout()
			root := cmd.Root()

			if len(args) == 0 {
				printCommandList(out, root)
				fmt.Fprintln(out)
				printConfigKeyList(out)
				return nil
			}

			topic := args[0]

			// 1) comando conhecido → ajuda do comando.
			if sub, _, err := root.Find([]string{topic}); err == nil && sub != root {
				sub.SetOut(out)
				sub.InitDefaultHelpFlag()
				return sub.Help()
			}

			// 2) chave de configuração conhecida → documentação da chave.
			for _, d := range configDocs {
				if d.Key == topic {
					printConfigKeyDoc(out, d)
					return nil
				}
			}

			// 3) assunto desconhecido → erro com sugestão útil.
			msg := "assunto desconhecido: " + topic
			if s := suggestTopic(topic, root); s != "" {
				msg += "\nVocê quis dizer: " + s + "?"
			}
			return errors.New(msg)
		},
	}
}

File: npm/src/commands/help.js
'use strict'
const { Command } = require('commander')

const configDocs = {
  adr_dirs: {
    type: 'list of strings',
    default: '["docs/adr"]',
    description: 'Diretórios onde os ADRs são armazenados.',
    example: 'adr_dirs:\n  - docs/adr\n  - docs/adr/zeus',
    impact: 'Todos os diretórios listados são varridos na validação de ADRs.'
  },
  req_dir: {
    type: 'string',
    default: '"docs/req"',
    description: 'Diretório onde as REQs são armazenadas.',
    example: 'req_dir: docs/requisicoes',
    impact: 'Altera onde o trackfw busca e cria arquivos de REQ.'
  },
  roadmap_dir: {
    type: 'string',
    default: '"docs/roadmaps"',
    description: 'Diretório raiz dos roadmaps.',
    example: 'roadmap_dir: docs/roadmaps',
    impact: 'Subtrees backlog/, wip/, blocked/, done/, abandoned/ são relativos a este diretório.'
  },
  roadmap_namespacing: {
    type: 'flat|by_agent',
    default: '"flat"',
    description: 'Estratégia de namespacing dos roadmaps.',
    example: 'roadmap_namespacing: by_agent',
    impact: 'Com by_agent, roadmaps ficam em subpastas por agente (ex: wip/apolo/RM-001.md).'
  },
  agents: {
    type: 'list of strings',
    default: '[]',
    description: 'Lista de agentes ativos no projeto.',
    example: 'agents:\n  - apolo\n  - afrodite',
    impact: 'Agentes listados têm suas subpastas criadas automaticamente em roadmap_dir.'
  },
  governance_mode: {
    type: 'string',
    default: '""',
    description: 'Modo de governança (strict, lenient).',
    example: 'governance_mode: strict',
    impact: 'strict bloqueia commits/merges com violations; lenient emite apenas avisos.'
  },
  lenient_until: {
    type: 'date (YYYY-MM-DD)',
    default: '""',
    description: 'Data até quando o modo lenient está ativo.',
    example: 'lenient_until: 2026-12-31',
    impact: 'Após a data, o modo volta automaticamente para strict.'
  },
  wip_limit: {
    type: 'integer',
    default: '1',
    description: 'Limite de itens WIP simultâneos.',
    example: 'wip_limit: 3',
    impact: 'Aumentar reduz a frequência de bloqueio.'
  },
  wip_by_squad: {
    type: 'boolean',
    default: 'false',
    description: 'Aplicar limite WIP por squad individualmente.',
    example: 'wip_by_squad: true',
    impact: 'Cada squad tem seu próprio contador de WIP em vez de um limite global.'
  },
  require_req_in_commit: {
    type: 'boolean',
    default: 'false',
    description: 'Exigir referência de REQ em mensagens de commit.',
    example: 'require_req_in_commit: true',
    impact: 'Commits sem menção a uma REQ são rejeitados pelo hook de pre-commit.'
  },
  'link_fields.req': {
    type: 'list of strings',
    default: '["REQ:"]',
    description: 'Marcadores que identificam link a REQ.',
    example: 'link_fields:\n  req:\n    - "REQ:"\n    - "Requisito:"',
    impact: 'Qualquer marcador listado é aceito para detectar vínculo com REQ.'
  },
  'link_fields.adr': {
    type: 'list of strings',
    default: '["ADR:"]',
    description: 'Marcadores que identificam link a ADR.',
    example: 'link_fields:\n  adr:\n    - "ADR:"\n    - "Decision:"',
    impact: 'Qualquer marcador listado é aceito para detectar vínculo com ADR.'
  },
  'link_fields.roadmap': {
    type: 'list of strings',
    default: '["Roadmap:"]',
    description: 'Marcadores que identificam link a Roadmap.',
    example: 'link_fields:\n  roadmap:\n    - "Roadmap:"',
    impact: 'Qualquer marcador listado é aceito para detectar vínculo com Roadmap.'
  },
  acceptance_markers: {
    type: 'list of strings',
    default: '["## Acceptance Criteria", "## Critérios de Aceite"]',
    description: 'Marcadores de critério de aceite.',
    example: 'acceptance_markers:\n  - "## Acceptance Criteria"\n  - "## AC"',
    impact: 'Roadmaps WIP sem nenhum desses marcadores disparam a regra wip_acceptance.'
  },
  'rules.wip_has_req': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: WIP sem REQ linkada.',
    example: 'rules:\n  wip_has_req: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.wip_acceptance': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: WIP sem critérios de aceite.',
    example: 'rules:\n  wip_acceptance: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.wip_limit': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: excesso de itens WIP.',
    example: 'rules:\n  wip_limit: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.stale_wip': {
    type: 'off|warning|error',
    default: '"warning"',
    description: 'Severidade: WIP sem atualização recente.',
    example: 'rules:\n  stale_wip: error',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.adr_orphan': {
    type: 'off|warning|error',
    default: '"warning"',
    description: 'Severidade: ADR sem REQ vinculada.',
    example: 'rules:\n  adr_orphan: error',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.ref_targets_exist': {
    type: 'off|warning|error',
    default: '"warning"',
    description: 'Severidade: referências com destino inexistente.',
    example: 'rules:\n  ref_targets_exist: error',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.folder_status': {
    type: 'off|warning|error',
    default: '"warning"',
    description: 'Severidade: coerência entre pasta e status do arquivo.',
    example: 'rules:\n  folder_status: error',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.filename_uniqueness': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: nomes de arquivo duplicados.',
    example: 'rules:\n  filename_uniqueness: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.blocked_by_draft_adr': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: REQ bloqueada por ADR em rascunho.',
    example: 'rules:\n  blocked_by_draft_adr: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  trace_id_field: {
    type: 'string',
    default: '""',
    description: 'Campo de frontmatter usado como trace ID para verificação bidirecional REQ↔Roadmap. Vazio desativa a verificação.',
    example: 'trace_id_field: req_id',
    impact: 'Quando configurado, ativa as regras traceid_* que garantem que toda REQ tem Roadmap correspondente e vice-versa.'
  },
  'rules.traceid_orphan_roadmap': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: Roadmap com trace ID sem REQ correspondente.',
    example: 'rules:\n  traceid_orphan_roadmap: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.traceid_orphan_req': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: REQ com trace ID sem Roadmap correspondente.',
    example: 'rules:\n  traceid_orphan_req: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.traceid_state_mismatch': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: REQ e Roadmap com mesmo trace ID em estados diferentes (ex: REQ em done, Roadmap em wip).',
    example: 'rules:\n  traceid_state_mismatch: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.traceid_duplicate_req': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: mesmo trace ID aparece em mais de uma REQ.',
    example: 'rules:\n  traceid_duplicate_req: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  },
  'rules.traceid_duplicate_roadmap': {
    type: 'off|warning|error',
    default: '"error"',
    description: 'Severidade: mesmo trace ID aparece em mais de um Roadmap.',
    example: 'rules:\n  traceid_duplicate_roadmap: warning',
    impact: 'error bloqueia; warning apenas reporta; off desativa a regra.'
  }
}

/**
 * Retorna string com a listagem tabular de todas as keys configuráveis.
 * @returns {string}
 */
function listKeys() {
  const COL_KEY = 28
  const COL_DEFAULT = 36
  const header = 'KEY'.padEnd(COL_KEY) + 'DEFAULT'.padEnd(COL_DEFAULT) + 'DESCRIÇÃO'
  const sep = '─'.repeat(COL_KEY + COL_DEFAULT + 40)
  const lines = [header, sep]
  for (const [key, doc] of Object.entries(configDocs)) {
    lines.push(
      key.padEnd(COL_KEY) +
      doc.default.padEnd(COL_DEFAULT) +
      doc.description
    )
  }
  return lines.join('\n')
}

/**
 * Retorna string com a documentação detalhada de uma key.
 * Retorna null se a key não existir.
 * @param {string} key
 * @returns {string|null}
 */
function describeKey(key) {
  const doc = configDocs[key]
  if (!doc) return null
  return [
    key,
    `  Type:    ${doc.type}`,
    `  Default: ${doc.default}`,
    `  Desc:    ${doc.description}`,
    `  Example:`,
    ...doc.example.split('\n').map(l => `    ${l}`),
    `  Impact:  ${doc.impact}`
  ].join('\n')
}

const cmd = new Command('help')
cmd.description('Exibe documentação das keys configuráveis do trackfw.yaml')
cmd.argument('[key]', 'key específica para detalhar')
cmd.action((key) => {
  if (!key) {
    console.log(listKeys())
    return
  }
  const output = describeKey(key)
  if (!output) {
    console.error(`chave desconhecida: ${key}`)
    process.exit(1)
  }
  console.log(output)
})

module.exports = cmd
module.exports.listKeys = listKeys
module.exports.describeKey = describeKey
